services/blockchain-connector/shared/utils/parser.js

```diff
@@ -38,7 +40,10 @@ const parseInputBySchema = (input, schema) => {
 	if (typeof input !== 'object') {
 		if (schemaDataType === 'string') return String(input);
 		if (schemaDataType === 'boolean') return Boolean(input);
-		if (schemaDataType === 'bytes') return Buffer.from(input, 'hex');
+		if (schemaDataType === 'bytes') {
+			if (schema.format === 'lisk32') { return getAddressFromLisk32Address(input); }
+			return Buffer.from(input, 'hex');
```

Comment:
Curious, why do we convert hex to buffer here?

Reply:
Because the schema expects the value to be an instance of Buffer and in the serialized form the value is a hex string.